src/modules/menu/components/advancedTools/ImportFile/parser/KMLParser.class.js

```diff
@@ -0,0 +1,90 @@
+import { getFileFromUrl } from '@/api/files.api'
+import KMLLayer from '@/api/layers/KMLLayer.class'
+import EmptyFileContentError from '@/modules/menu/components/advancedTools/ImportFile/parser/errors/EmptyFileContentError.error'
+import InvalidFileContentError from '@/modules/menu/components/advancedTools/ImportFile/parser/errors/InvalidFileContentError.error'
+import OutOfBoundsError from '@/modules/menu/components/advancedTools/ImportFile/parser/errors/OutOfBoundsError.error'
+import FileParser from '@/modules/menu/components/advancedTools/ImportFile/parser/FileParser.class'
+import { WGS84 } from '@/utils/coordinates/coordinateSystems'
+import { getExtentIntersectionWithCurrentProjection } from '@/utils/extentUtils'
+import { getKmlExtent } from '@/utils/kmlUtils'
+
+/**
+ * Checks if file is KMLs
+ *
+ * @param {ArrayBuffer | String} fileContent
+ * @returns {boolean}
+ */
+export function isKml(fileContent) {
+    let stringValue = fileContent
+    if (fileContent instanceof ArrayBuffer) {
+        stringValue = new TextDecoder('utf-8').decode(fileContent)
+    }
+    return /^\s*(<\?xml\b[^>]*\?>)?\s*(<!--(.*?)-->\s*)*<(kml:)?kml\b[^>]*>[\s\S.]*<\/(kml:)?kml\s*>/g.test(
```

Comment:
I guess you could spare the regex hassle with simply this:
```
const parser = new DOMParser()
parser.parser.parseFromString("...", "application/xml")

Reply:
@schtibe `<kml:kml>` is a valid kml and we do have users using it.